===================================================================
RCS file: /cvsroot/ppp/pppd/main.c,v
retrieving revision 1.153
retrieving revision 1.154
diff -u -r1.153 -r1.154
--- ppp/pppd/main.c	2006/06/04 03:52:50	1.153
+++ ppp/pppd/main.c	2006/12/19 07:48:19	1.154
@@ -66,7 +66,7 @@
  * OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#define RCSID	"$Id: main.c,v 1.153 2006/06/04 03:52:50 paulus Exp $"
+#define RCSID	"$Id: main.c,v 1.154 2006/12/19 07:48:19 paulus Exp $"
 
 #include <stdio.h>
 #include <ctype.h>
@@ -1567,6 +1567,8 @@
 	if (errfd == 0 || errfd == 1)
 		errfd = dup(errfd);
 
+	closelog();
+
 	/* dup the in, out, err fds to 0, 1, 2 */
 	if (infd != 0)
 		dup2(infd, 0);
@@ -1575,7 +1577,6 @@
 	if (errfd != 2)
 		dup2(errfd, 2);
 
-	closelog();
 	if (log_to_fd > 2)
 		close(log_to_fd);
 	if (the_channel->close)
